tests/cpp/multidevice_transformer.cpp

```diff
@@ -0,0 +1,660 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <vector>
+
+#include <fusion.h>
+#include <ops/all_ops.h>
+#include <tests/cpp/multidevice.h>
+#include <tests/cpp/multidevice_transformer.h>
+
+namespace nvfuser {
+namespace {
+// TODO: These linear_backwards helper functions can be merged once
+// we do not have logically split rfactor domain.
+struct LinearBackwardsResult {
+  TensorView* grad_x;
+  TensorView* grad_w;
+  TensorView* grad_b;
+};
+
+// x format: [i0, i1] dtype
+// weight format: [DID(D), i2/D, i1] dtype
+// grad format: [DID(D) i0, i2/D] float or dtype
+// outputs: grad_x [i0, i1] dtype
+// grad_w [DID i2/D, i1] dtype
+// grad_b [DID i2/2] dtype
+LinearBackwardsResult linear_backwards(
```

Comment:
```suggestion
LinearBackwardsResult linearBackwards(
```
It's fine leaving it as is. For new code, I'd try to stick with the style convention.

Reply:
it's a good time to do little cleanups!